src/lib/FileManager.js

```diff
@@ -105,7 +105,13 @@ export default class FileManager {
   }
 
   async downloadFileFromDescriptor(fileDescriptor) {
+    var retries = 3;
     var integration = this.integrationManager.integrationForFileDescriptor(fileDescriptor);
+    while (retries-- > 0 && !integration) {
+      setTimeout(function() {
```

Comment:
setTimeout is non-blocking, in the sense that the code inside would not be waited on before the function returns. So in this case the code inside setTimeout would run 3 times immediately after an initial 1000ms. To get around this you would either need to wrap setTimeout in an async function and use await, or use a recursive approach.